### ISSUE TITLE
shouldn't commit loss be (x - q.detach())^2 ?

### ISSUE THREAD

**Issue by JiayouZhang** [lucidrains/vector-quantize-pytorch#214]
https://github.com/lucidrains/vector-quantize-pytorch/blob/9d2594b4f1a92186bd267c79e70c4aa820950115/vector_quantize_pytorch/vector_quantize_pytorch.py#L1140

The commitment loss should move the encoder output close to the quantization, following Eq3 in the original VQVAE paper (https://arxiv.org/pdf/1711.00937 )

**Comment by lucidrains** [comment]
@JiayouZhang yea, that's taken care of in [this section](https://github.com/lucidrains/vector-quantize-pytorch/blob/9d2594b4f1a92186bd267c79e70c4aa820950115/vector_quantize_pytorch/vector_quantize_pytorch.py#L1235)
